### PR TITLE
Hint/feature/coordinator admin

### PR DESCRIPTION
**Pull request by nshoes**
Solves #225, #226, and #409.

I may implement the show/edit actions for `dashboard#users`, waiting for an email from Sarah.

### CODE REVIEW

File: app/policies/user_policy.rb [hunk 1]
```diff
@@ -34,6 +34,14 @@ def permitted_attributes
     end
   end
 
+  def reports?
+    user.role.in? [User::COORDINATOR, User::ADMIN]
+  end
+
+  def download_report?
+    user.role.in? [User::COORDINATOR, User::ADMIN]
+  end
```
Comment (benjaminwood) [inline app/policies/user_policy.rb:43]: Now that we have a reports controller, perhaps we can have a report policy?

It's permissions would be based solely on the user like you have here, there is no Report object to pass in.

File: app/models/office.rb [hunk 2]
```diff
@@ -12,83 +14,105 @@ class Office < ApplicationRecord
...
 
   alias_attribute :to_s, :name
 
-  def self.claimed_shifts_by_office
-    with_claimed_shifts.group('offices.id')
+  def self.total_volunteer_hours_by_office(current_user, start_at, end_at)
+    with_claimed_shifts(current_user)
+      .then { |scope| filter_by_date_range(scope, start_at, end_at) } 
+      .group(:id, :name)
+      .sum('shifts.duration / 60.0')
```
Comment (nshoes) [inline app/models/office.rb:43]: I feel like these methods that gather report data for `Office` and `User` don't really belong in the model. Seems like a service class would be more ideal, thoughts @benjaminwood?

File: spec/requests/dashboard_spec.rb [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      let(:user) { create :user, role: 'social_worker' }
+      before { sign_in user }
+
+      it 'is successful' do
+        get dashboard_users_path
+  
+        expect(response).to redirect_to :root
+      end
+    end
+  end
```
Comment (nshoes) [inline spec/requests/dashboard_spec.rb:96]: @benjaminwood I don't like this repeating pattern for checking authorization, but I dislike _not_ explicitly testing it more. In this case where we have a policy and spec that covers these scenarios already, is it worth to check at the controller level as well? My gut says yes.

**Comment by nshoes** [comment]
After testing on staging, I found an issue with formatting in the downloaded reports. Looking into it now.

**Comment by benjaminwood** [comment]
Looking good. A couple of things:

Can we add an "invitation status" column to the user's index view?

How about a min-height on report boxes so that "By User/ Volunteer Hours" is not shorter than the rest of the boxes when displayed at full width?

**Comment by nshoes** [comment]
> Can we add an "invitation status" column to the user's index view?
> 
> How about a min-height on report boxes so that "By User/ Volunteer Hours" is not shorter than the rest of the boxes when displayed at full width?

Done! Deployed to staging. Also fixed the CSV formatting bug.

File: app/views/dashboard/users.html.haml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    %thead
+      %tr
+        %th Name
+        %th Email
+        %th Role
+        -# %th Actions
+    %tbody
+      - users.each do |user|
+        %tr
+          %td= "#{user.first_name} #{user.last_name}"
```
Comment (nvick) [inline app/views/dashboard/users.html.haml:15]:
```suggestion
          %td= "#{user.to_s}”
```

File: app/views/dashboard/users.html.haml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+          %td= user.email
+          %td= user.role
```
Comment (nvick) [inline app/views/dashboard/users.html.haml:17]:
```suggestion
          %td= user.role_display
```

File: app/views/dashboard/users.html.haml (hunk 4 above)
Comment (dinjas) [reply app/views/dashboard/users.html.haml:15]: `to_s` isn't needed when you're interpolating - could just be ` %td= "#{user}”` or ` %td= user.to_s`

Comment (nvick) [reply app/views/dashboard/users.html.haml:15]: True and fixed

**Pull request merged**